src/parser/DASHTree.cpp

```diff
@@ -1841,7 +1841,8 @@ void DASHTree::RefreshLiveSegments()
                         }
                         if (segment.range_begin_ == search_pts)
                           break;
-                        else if (segment.range_begin_ > search_pts)
+                        else if (segment.range_begin_ > search_pts &&
```

Comment:
i cannot do a review of the change but just code style:
IMO this:
`std::addressof(*std::cbegin(repr->segments_.data)) != &segment`
can be simplified with
`&repr->segments_.data.front() != &segment`

Reply:
Yeah, thanks for the suggestion. I totally forgot about front method. I've pushed the new revision with changes mentioned in my last comment. The previous revision causes audio sync errors for some reason. To be honest I have no idea what it changes under the hood - I did it by trial and error method.